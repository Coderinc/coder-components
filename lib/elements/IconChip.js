import React from 'react';
import styled from 'styled-components';

const ChipWrapper = styled.div`
  background-color: ${props => props.theme.activeBackground};
  color: ${props => props.theme.primaryFont};

  padding: 0.375rem 0.5rem 0.25rem;
  margin: 0.25rem;
  border-radius: 0.125rem;

  font-size: 0.625rem;
  font-weight: lighter;
  letter-spacing: 0.0625rem;

  display: flex;

  img {
    width: 0.875rem;
    height: 0.875rem;
    border-radius: 50%;
    margin-right: 0.375rem;
  }
`;

export const IconChip = ({ icon, label, ...props }) => (
  <ChipWrapper {...props}>
    <img
      alt="Icon"
      src={icon}
    />
    {label}
  </ChipWrapper>
);

export default IconChip;
