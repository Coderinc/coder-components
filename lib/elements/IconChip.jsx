import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ChipWrapper = styled.div`
  background-color: ${({ theme }) => theme.activeBackground};
  color: ${({ theme }) => theme.primaryFont};

  padding: 0.375rem 0.5rem 0.25rem;
  margin: 0.25rem;
  border-radius: 0.125rem;

  font-size: 0.625rem;
  font-weight: lighter;
  letter-spacing: 0.0625rem;

  display: flex;

  img {
    width: 0.875rem;
    height: 0.875rem;
    border-radius: 50%;
    margin-right: 0.375rem;
  }
`;

const IconChip = ({ icon, label, ...props }) => (
  <ChipWrapper {...props}>
    {icon && <img alt="Icon" src={icon} />}
    {label}
  </ChipWrapper>
);

IconChip.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string,
};

IconChip.defaultProps = {
  icon: null,
  label: 'Joe Schmoe',
};

export default IconChip;
