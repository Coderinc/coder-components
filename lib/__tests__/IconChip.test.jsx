import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import IconChip from '../elements/IconChip';

describe('IconChip', () => {
  it('renders correctly', () => {
    const component = renderer.create(<IconChip label="Test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});