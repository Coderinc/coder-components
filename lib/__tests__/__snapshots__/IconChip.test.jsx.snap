// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconChip renders correctly 1`] = `
.c0 {
  padding: 0.375rem 0.5rem 0.25rem;
  margin: 0.25rem;
  border-radius: 0.125rem;
  font-size: 0.625rem;
  font-weight: lighter;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 img {
  width: 0.875rem;
  height: 0.875rem;
  border-radius: 50%;
  margin-right: 0.375rem;
}

<div
  className="c0"
>
  Test
</div>
`;
